### PR TITLE
Thesahindia/UI/confirmView

### PR DESCRIPTION
**Pull request by thesahindia**
### Details

- Made confirmView component to reuse in confirmModal and confirmPopover
- Made delete confirmModal consistent
### Fixed Issues

$ https://github.com/Expensify/App/issues/7545

### Tests

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Open app
2. Navigate to Settings > Payments and try to delete any payment
3. Navigate to Settings > Workspace and try to delete the workspace
4. Navigate to a chat > Try to delete a comment
5. Verify the confirmModal/Popover is consistent
- [x] Verify that no errors appear in the JS console

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

Screenshot 2022-02-21 at 6 58 13 PM
Screenshot 2022-02-21 at 7 00 44 PM

Screenshot 2022-02-21 at 8 06 03 PM

#### Mobile Web
Screenshot 2022-02-21 at 6 49 56 PM
Screenshot 2022-02-21 at 8 35 02 PM
Screenshot 2022-02-21 at 9 50 03 PM

#### Desktop

Screenshot 2022-02-21 at 9 32 18 PM
Screenshot 2022-02-21 at 9 32 31 PM
Screenshot 2022-02-21 at 9 32 55 PM

#### iOS
Screenshot 2022-02-21 at 8 22 51 PM

Screenshot 2022-02-21 at 6 51 49 PM

#### Android

Screenshot 2022-02-21 at 6 45 01 PM
Screenshot 2022-02-21 at 6 45 36 PM
Screenshot 2022-02-21 at 8 13 50 PM

### CODE REVIEW

**Comment by parasharrajat** [comment]
Testing shortly...

File: src/components/ConfirmView.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        <View style={[styles.flexRow, styles.mb4]}>
+            <Header title={props.title} />
+        </View>
+
+        {_.isString(props.prompt)
+            ? (
+                <Text>
+                    {props.prompt}
+                </Text>
+            ) : (props.prompt)}
```
Comment (parasharrajat) [inline src/components/ConfirmView.js:68]:
```suggestion
            ) : props.prompt}
```

File: src/components/ConfirmView.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    cancelText: '',
+    prompt: '',
+    success: true,
+    danger: false,
+    onCancel: () => {},
+    shouldShowCancelButton: true,
+    viewStyles: [],
+};
+
+const ConfirmView = props => (
```
Comment (parasharrajat) [inline src/components/ConfirmView.js:57]:
```suggestion
const ConfirmContent = props => (
```

File: src/components/ConfirmView.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    success: PropTypes.bool,
+
+    /** Is the action destructive */
+    danger: PropTypes.bool,
+
+    /** Whether we should show the cancel button */
+    shouldShowCancelButton: PropTypes.bool,
+
+    /** Styles for view */
+    viewStyles: PropTypes.arrayOf(PropTypes.object),
```
Comment (parasharrajat) [inline src/components/ConfirmView.js:40]:
```suggestion
    contentStyles: PropTypes.arrayOf(PropTypes.object),
```

**Review by parasharrajat — CHANGES_REQUESTED** [review]

File: src/components/ConfirmPopover.js [hunk 4]
```diff
@@ -2,7 +2,8 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import Popover from './Popover';
 import withWindowDimensions, {windowDimensionsPropTypes} from './withWindowDimensions';
-import ConfirmView from './ConfirmView';
+import ConfirmContent from './ConfirmContent';
+
```
Comment (parasharrajat) [inline src/components/ConfirmPopover.js:6]:
```suggestion
```

**Review by parasharrajat — CHANGES_REQUESTED** [review]

**Review by parasharrajat — DISMISSED** [review]
LGTM. 

@Beamanator 

:ribbon: :eyes: :ribbon:  C+ reviewed

**Comment by parasharrajat** [comment]
cc: @Beamanator

**Comment by Beamanator** [comment]
Thanks @parasharrajat I'll check this out today, I was on vacation last week :D

File: src/components/ConfirmContent.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    danger: PropTypes.bool,
+
+    /** Whether we should show the cancel button */
+    shouldShowCancelButton: PropTypes.bool,
+
+    /** Styles for view */
+    contentStyles: PropTypes.arrayOf(PropTypes.object),
+
+    ...withLocalizePropTypes,
+
```
Comment (Beamanator) [inline src/components/ConfirmContent.js:43]:
```suggestion
```

File: src/components/ConfirmContent.js [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+            text={props.confirmText || props.translate('common.yes')}
+        />
+        {props.shouldShowCancelButton
+    && (
+        <Button
+            style={[styles.mt3]}
+            onPress={props.onCancel}
+            text={props.cancelText || props.translate('common.no')}
+        />
+    )}
```
Comment (Beamanator) [inline src/components/ConfirmContent.js:85]:
```suggestion
        {props.shouldShowCancelButton && (
            <Button
                style={[styles.mt3]}
                onPress={props.onCancel}
                text={props.cancelText || props.translate('common.no')}
            />
        )}
```

File: src/components/ConfirmContent.js [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    /** Cancel button text */
+    cancelText: PropTypes.string,
+
+    /** Modal content text/element */
+    prompt: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+
+    /** Whether we should use the success button color */
+    success: PropTypes.bool,
+
+    /** Is the action destructive */
```
Comment (Beamanator) [inline src/components/ConfirmContent.js:33]: What about this?

```suggestion
    /** Whether we should use the danger button color. Use if the action is destructive */
```

**Review by Beamanator — CHANGES_REQUESTED** [review]
Overall looks great, just a few style changes and comment change suggestion

**Review by Beamanator — APPROVED** [review]

**Comment by Beamanator** [comment]
Thanks for the quick edits @thesahindia ! 👍

**Comment by mvtglobally** [comment]
We are still able to repro https://github.com/Expensify/App/issues/7768 in IOS, so limited QA possible. Rest platforms ok

**Pull request merged**